src/client.js

```diff
@@ -17,15 +17,20 @@ const store = configureStore(initialState, baseHistory)
 const history = syncHistoryWithStore(baseHistory, store)
 const root = document.getElementById('app')
 
-const renderApp = () => (
+const renderApp = (renderProps) => (
   <AppContainer>
     <Provider store={store}>
-      <Router history={history} routes={routes} />
+      <Router {...renderProps} />
     </Provider>
   </AppContainer>
 )
```

Comment:
Is the `match` call necessary here? I wonder if just placing `key={Math.random()}` solves HMR?